File: openharmony/arkcompiler_runtime_core/static_core/plugins/ets/tests/interop_js/tests/generic_as_parameter/ts_to_sts/generic_as_parameter.js
'use strict';
/**
 * Copyright (c) 2024 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the 'License');
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an 'AS IS' BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, '__esModule', { value: true });
exports.DataClass = exports.userClassInstance = exports.UserClass = 
exports.GTypeClass = exports.GInterfaceClass = 
exports.extendGenericLiteral = exports.extendGenericTuple = 
exports.extendGenericUnion = exports.extendGenericObj = 
exports.extendGenericArr = exports.extendGenericBool = 
exports.extendGenericString = exports.extendGenericNumber = 
exports.genericTypeFunctionReturnAny = exports.tsLiteral = exports.tsTuple = 
exports.tsUnion = exports.tsObj = exports.tsArr = exports.tsBool = 
exports.tsString = exports.tsNumber = void 0;
exports.anyTypeParameter = anyTypeParameter;
exports.anyTypeParameterExplicitCallFromTsInt = anyTypeParameterExplicitCallFromTsInt;
exports.anyTypeParameterExplicitCallFromTsString = anyTypeParameterExplicitCallFromTsString;
exports.anyTypeParameterExplicitCallFromTsBool = anyTypeParameterExplicitCallFromTsBool;
exports.anyTypeParameterExplicitCallFromTsArr = anyTypeParameterExplicitCallFromTsArr;
exports.anyTypeParameterExplicitCallFromTsObj = anyTypeParameterExplicitCallFromTsObj;
exports.anyTypeParameterExplicitCallFromTsUnion = anyTypeParameterExplicitCallFromTsUnion;
exports.anyTypeParameterExplicitCallFromTsTuple = anyTypeParameterExplicitCallFromTsTuple;
exports.anyTypeParameterExplicitCallFromTsLiteral = anyTypeParameterExplicitCallFromTsLiteral;
exports.genericTypeFunctionExplicitCallFromTsInt = genericTypeFunctionExplicitCallFromTsInt;
exports.genericTypeFunctionExplicitCallFromTsString = genericTypeFunctionExplicitCallFromTsString;
exports.genericTypeFunctionExplicitCallFromTsBool = genericTypeFunctionExplicitCallFromTsBool;
exports.genericTypeFunctionExplicitCallFromTsArr = genericTypeFunctionExplicitCallFromTsArr;
exports.genericTypeFunctionExplicitCallFromTsObj = genericTypeFunctionExplicitCallFromTsObj;
exports.genericTypeFunctionExplicitCallFromTsUnion = genericTypeFunctionExplicitCallFromTsUnion;
exports.genericTypeFunctionExplicitCallFromTsTuple = genericTypeFunctionExplicitCallFromTsTuple;
exports.genericTypeFunctionExplicitCallFromTsLiteral = genericTypeFunctionExplicitCallFromTsLiteral;
exports.genericExtendInterface = genericExtendInterface;
exports.genericExtendType = genericExtendType;
exports.tupleGeneric = tupleGeneric;
exports.collectGeneric = collectGeneric;
exports.createClassFromTs = createClassFromTs;
exports.genericDefaultInt = genericDefaultInt;
exports.genericDefaultString = genericDefaultString;
exports.genericDefaultBool = genericDefaultBool;
exports.genericDefaultArr = genericDefaultArr;
exports.genericDefaultObj = genericDefaultObj;
exports.genericDefaultUnion = genericDefaultUnion;
exports.genericDefaultTuple = genericDefaultTuple;
exports.genericDefaultLiteral = genericDefaultLiteral;
exports.genericDefaultIntCallFromTs = genericDefaultIntCallFromTs;
exports.genericDefaultStringCallFromTs = genericDefaultStringCallFromTs;
exports.genericDefaultBoolCallFromTs = genericDefaultBoolCallFromTs;
exports.genericDefaultArrCallFromTs = genericDefaultArrCallFromTs;
exports.genericDefaultObjCallFromTs = genericDefaultObjCallFromTs;
exports.genericDefaultUnionCallFromTs = genericDefaultUnionCallFromTs;
exports.genericDefaultTupleCallFromTs = genericDefaultTupleCallFromTs;
exports.genericDefaultLiteralCallFromTs = genericDefaultLiteralCallFromTs;
exports.genericExtendClass = genericExtendClass;
exports.tsNumber = 1;
exports.tsString = 'string';
exports.tsBool = true;
exports.tsArr = [exports.tsNumber];
exports.tsObj = {};
exports.tsUnion = exports.tsNumber;
exports.tsTuple = [exports.tsNumber, exports.tsBool];
exports.tsLiteral = exports.tsNumber;
function anyTypeParameter(arg) {
    return arg;
}
function anyTypeParameterExplicitCallFromTsInt() {
    return anyTypeParameter(exports.tsNumber);
}
function anyTypeParameterExplicitCallFromTsString() {
    return anyTypeParameter(exports.tsString);
}
function anyTypeParameterExplicitCallFromTsBool() {
    return anyTypeParameter(exports.tsBool);
}
function anyTypeParameterExplicitCallFromTsArr() {
    return anyTypeParameter(exports.tsArr);
}
function anyTypeParameterExplicitCallFromTsObj() {
    return anyTypeParameter(exports.tsObj);
}
function anyTypeParameterExplicitCallFromTsUnion() {
    return anyTypeParameter(exports.tsUnion);
}
function anyTypeParameterExplicitCallFromTsTuple() {
    return anyTypeParameter(exports.tsTuple);
}
function anyTypeParameterExplicitCallFromTsLiteral() {
    return anyTypeParameter(exports.tsLiteral);
}
let genericTypeFunctionReturnAny = function (arg) {
    return arg;
};
exports.genericTypeFunctionReturnAny = genericTypeFunctionReturnAny;
function genericTypeFunctionExplicitCallFromTsInt() {
    return (0, exports.genericTypeFunctionReturnAny)(exports.tsNumber);
}
function genericTypeFunctionExplicitCallFromTsString() {
    return (0, exports.genericTypeFunctionReturnAny)(exports.tsString);
}
function genericTypeFunctionExplicitCallFromTsBool() {
    return (0, exports.genericTypeFunctionReturnAny)(exports.tsBool);
}
function genericTypeFunctionExplicitCallFromTsArr() {
    return (0, exports.genericTypeFunctionReturnAny)(exports.tsArr);
}
function genericTypeFunctionExplicitCallFromTsObj() {
    return (0, exports.genericTypeFunctionReturnAny)(exports.tsObj);
}
function genericTypeFunctionExplicitCallFromTsUnion() {
    return (0, exports.genericTypeFunctionReturnAny)(exports.tsUnion);
}
function genericTypeFunctionExplicitCallFromTsTuple() {
    return (0, exports.genericTypeFunctionReturnAny)(exports.tsTuple);
}
function genericTypeFunctionExplicitCallFromTsLiteral() {
    return (0, exports.genericTypeFunctionReturnAny)(exports.tsLiteral);
}
let extendGenericNumber = function (arg) { return arg; };
exports.extendGenericNumber = extendGenericNumber;
let extendGenericString = function (arg) { return arg; };
exports.extendGenericString = extendGenericString;
let extendGenericBool = function (arg) { return arg; };
exports.extendGenericBool = extendGenericBool;
let extendGenericArr = function (arg) { return arg; };
exports.extendGenericArr = extendGenericArr;
let extendGenericObj = function (arg) { return arg; };
exports.extendGenericObj = extendGenericObj;
let extendGenericUnion = function (arg) { return arg; };
exports.extendGenericUnion = extendGenericUnion;
let extendGenericTuple = function (arg) { return arg; };
exports.extendGenericTuple = extendGenericTuple;
let extendGenericLiteral = function (arg) { return arg; };
exports.extendGenericLiteral = extendGenericLiteral;
let GInterfaceClass = /** @class */ (function () {
    function gInterfaceClass() {
        this.data = 'string';
    }
    return gInterfaceClass;
}());
exports.GInterfaceClass = GInterfaceClass;
let GTypeClass = /** @class */ (function () {
    function gTypeClass() {
        this.data = 'string';
    }
    return gTypeClass;
}());
exports.GTypeClass = GTypeClass;
function genericExtendInterface(arg) {
    return arg;
}
function genericExtendType(arg) {
    return arg;
}
function tupleGeneric(arg) {
    return [arg[1], arg[0]];
}
function collectGeneric(...arg) {
    let args = [];
    for (let _i = 0; _i < arg.length; _i++) {
        args[_i] = arg[_i];
    }
    return args;
}
let UserClass = /** @class */ (function () {
    function userClass(name, age) {
        this.name = name;
        this.age = age;
    }
    userClass.prototype.getName = function () {
        return this.name;
    };
    userClass.prototype.getAge = function () {
        return this.age;
    };
    return userClass;
}());
exports.UserClass = UserClass;
function createClassFromTs() {
    return new UserClass(exports.tsString, exports.tsNumber);
}
exports.userClassInstance = new UserClass(exports.tsString, exports.tsNumber);
function genericDefaultInt(arg) {
    return arg;
}
function genericDefaultString(arg) {
    return arg;
}
function genericDefaultBool(arg) {
    return arg;
}
function genericDefaultArr(arg) {
    return arg;
}
function genericDefaultObj(arg) {
    return arg;
}
function genericDefaultUnion(arg) {
    return arg;
}
function genericDefaultTuple(arg) {
    return arg;
}
function genericDefaultLiteral(arg) {
    return arg;
}
function genericDefaultIntCallFromTs() {
    return genericDefaultInt(exports.tsNumber);
}
function genericDefaultStringCallFromTs() {
    return genericDefaultString(exports.tsString);
}
function genericDefaultBoolCallFromTs() {
    return genericDefaultBool(exports.tsBool);
}
function genericDefaultArrCallFromTs() {
    return genericDefaultArr(exports.tsArr);
}
function genericDefaultObjCallFromTs() {
    return genericDefaultObj(exports.tsObj);
}
function genericDefaultUnionCallFromTs() {
    return genericDefaultUnion(exports.tsUnion);
}
function genericDefaultTupleCallFromTs() {
    return genericDefaultTuple(exports.tsTuple);
}
function genericDefaultLiteralCallFromTs() {
    return genericDefaultLiteral(exports.tsLiteral);
}
let DataClass = /** @class */ (function () {
    function dataClass() {
        this.data = 'string';
    }
    return dataClass;
}());
exports.DataClass = DataClass;
function genericExtendClass(arg) {
    return arg;
}

File: openharmony/arkcompiler_runtime_core/static_core/plugins/ets/tests/interop_js/tests/generic_as_parameter/ts_to_sts/generic_as_parameter.cpp
/**
 * Copyright (c) 2024 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include <gtest/gtest.h>
#include "ets_interop_js_gtest.h"

namespace ark::ets::interop::js::testing {

class EtsGenericAsParameterTsToEtsTest : public EtsInteropTest {};

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterInt)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterInt"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterString)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterString"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterBool)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterBool"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterArr)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterArr"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterObj)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterObj"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterUnion)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterUnion"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterTuple)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterTuple"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterExplicitCallInt)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterExplicitCallInt"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterExplicitCallString)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterExplicitCallString"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterExplicitCallBool)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterExplicitCallBool"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterExplicitCallArr)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterExplicitCallArr"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterExplicitCallObj)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterExplicitCallObj"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterExplicitCallUnion)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterExplicitCallUnion"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterExplicitCallTuple)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterExplicitCallTuple"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkAnyTypeParameterExplicitCallLiteral)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkAnyTypeParameterExplicitCallLiteral"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionReturnAnyInt)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionReturnAnyInt"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionReturnAnyString)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionReturnAnyString"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionReturnAnyBool)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionReturnAnyBool"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionReturnAnyArr)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionReturnAnyArr"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionReturnAnyObj)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionReturnAnyObj"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionReturnAnyUnion)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionReturnAnyUnion"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionReturnAnyTuple)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionReturnAnyTuple"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionExplicitCallInt)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionExplicitCallInt"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionExplicitCallString)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionExplicitCallString"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionExplicitCallBool)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionExplicitCallBool"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionExplicitCallArr)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionExplicitCallArr"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionExplicitCallObj)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionExplicitCallObj"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionExplicitCallUnion)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionExplicitCallUnion"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionExplicitCallTuple)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionExplicitCallTuple"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericTypeFunctionExplicitCallLiteral)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericTypeFunctionExplicitCallLiteral"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkExtendGenericNumber)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkExtendGenericNumber"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkExtendGenericString)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkExtendGenericString"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkExtendGenericBool)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkExtendGenericBool"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkExtendGenericArr)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkExtendGenericArr"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkExtendGenericObj)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkExtendGenericObj"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkExtendGenericUnion)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkExtendGenericUnion"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkExtendGenericTuple)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkExtendGenericTuple"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkExtendGenericLiteral)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkExtendGenericLiteral"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericExtendInterface)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericExtendInterface"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericExtendType)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericExtendType"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkTupleGeneric)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkTupleGeneric"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkCollectGenericInt)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkCollectGenericInt"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkCollectGenericString)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkCollectGenericString"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkCollectGenericBool)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkCollectGenericBool"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkCollectGenericObj)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkCollectGenericObj"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericUserClass)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericUserClass"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkCreateClassFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkCreateClassFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkUserClassInstance)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkUserClassInstance"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultInt)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultInt"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultString)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultString"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultBool)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultBool"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultArr)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultArr"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultObj)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultObj"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultUnion)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultUnion"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultTuple)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultTuple"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultLiteral)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultLiteral"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultIntCallFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultIntCallFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultStringCallFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultStringCallFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultBoolCallFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultBoolCallFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultArrCallFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultArrCallFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultObjCallFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultObjCallFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultUnionCallFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultUnionCallFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultTupleCallFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultTupleCallFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericDefaultLiteralCallFromTs)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericDefaultLiteralCallFromTs"));
}

TEST_F(EtsGenericAsParameterTsToEtsTest, checkGenericExtendClass)
{
    ASSERT_EQ(true, CallEtsMethod<bool>("checkGenericExtendClass"));
}

}  // namespace ark::ets::interop::js::testing
